### PR TITLE
my first pdns script

### PR DESCRIPTION
**Pull request by nj814**
.. to block malicious DNS domains

### Description of Change
**add file to repository** 

### Test results

- Did you do any localized testing?

### Who should know about the change?

/cc: @githubteacher

### CODE REVIEW

**Review by nj814 — COMMENTED** [review]
"com.", "top.", "tk." get blocked by NXDOMAIN answer

File: nxscript.lua [hunk 1]
```diff
@@ -0,0 +1,33 @@
+blockset = newDS()
+blockset:add{"powerdns.org", "xxx"}
+
+dropset = newDS();
+dropset:add("web.de")
+
+malwareset = newDS()
+malwareset:add{"com.", "top.", "tk."}
```
Comment (nj814) [inline nxscript.lua:8]: Line to add blocked domains

File: nxscript.lua [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+function preresolve(dq)
+
+  pdnslog("prereesolve handler called for: "..dq.remoteaddr:toString().. ", local: ".. dq.localaddr:toString()..", ".. dq.qname:toString()..", ".. dq.qtype)
+  
+  if dropset:check(dq.qname) then
+    dq.rcode = pdns.DROP  
+    return true;
+  end
+
+  if malwareset:check(dq.qname) then
```
Comment (nj814) [inline nxscript.lua:21]: set responce, send to the request

Comment (nj814) [inline nxscript.lua:21]: ...

**Review by nj814 — COMMENTED** [review]
MY docu to the code

File: nxscript.lua [hunk 3]
```diff
@@ -0,0 +1,33 @@
+blockset = newDS()
+blockset:add{"powerdns.org", "xxx"}
+
+dropset = newDS();
+dropset:add("web.de")
+
```
Comment (szigorferihegyi) [inline nxscript.lua:6]: `This is my favorite line in the whole code!`
:+1:

**Review by szigorferihegyi — APPROVED** [review]
Hereby I approve the production implementation of this code on each and every device.

**Pull request merged**